.travis/travis.py

```diff
@@ -16,95 +48,199 @@ def test_diffs(diffs):
         test_diffs(diffs)
 
-        target = repo.branches[TRAVIS_BRANCH].commit
+        target = repo.branches[GIT_BRANCH].commit
         pull_request = repo.head.commit
-        pr_diffs = target.diff(pull_request, paths=[docker_tag_dir])
+        pr_diffs = target.diff(pull_request, paths=[hub_tag_dir])
 
         if pr_diffs:
             # Test that the dockerfiles build
-            test_builds(docker_tag_dir)
+            test_builds(hub_tag_dir)
 
     else:
-        print("Testing CronJob for Branch: ", TRAVIS_BRANCH)
+        # If this is a test from CronJob
+        print("Testing CronJob for Branch: ", GIT_BRANCH)
 
-        # Test that dockerfile generation has changed nothing
-        # and that all dockerfiles are up to date
         try:
+            # Test that dockerfile generation has changed nothing
+            # and that all dockerfiles are up to date
             test_diffs(diffs)
         except ValueError as err:
-            # TODO: auto make a PR on github with new patch
+            # If there are changes, only proceed for the deafult branch
+            if GIT_BRANCH == GIT_DEFAULT_BRANCH:
+
+                # Initlise github interfaces
+                g = github.Github(login_or_token=GIT_TOKEN)
+                g_origin_repo = g.get_repo(
+                    full_name_or_id=GIT_ORIGIN_REPO_SLUG)
+                g_upstream_repo = g.get_repo(
+                    full_name_or_id=GIT_UPSTREAM_REPO_SLUG)
+
+                # Define common attributes for new PR
+                pr_branch_name = (
+                    '/').join([HUB_REPO, HUB_RELEASE, HUB_OS_NAME, HUB_OS_CODE_NAME])
+                pr_head_name = GIT_USER + ':' + pr_branch_name
+
+                pr_remote = git.remote.Remote(repo=repo, name='origin')
+                pr_remote.add(repo=repo, name='upstream_pr',
+                              url=GIT_REMOTE_ORIGIN_TOKEN_URL)
+
+                # Commit changes to Dockerfiles
+                repo.git.add(all=True)
+                message = "Updating Dockerfiles\n" + \
+                    "This is an automated CI commit"
+                repo.git.commit(m=message, author=GIT_AUTHOR)
+
+                # Update the Docker Library
+                manifest = os.path.join(hub_repo_dir, 'manifest.yaml')
+                output = os.path.join(hub_repo_dir, HUB_REPO)
+                create_dockerlibrary = import_create_dockerlibrary(
+                    hub_repo_dir)
+                create_dockerlibrary.main((
+                    '--manifest', manifest,
+                    '--output', output))
+
+                # Commit changes to Docker Library
+                repo.git.add(all=True)
+                message = "Updating Docker Library\n" + \
+                    "This is an automated CI commit"
+                repo.git.commit(m=message, author=GIT_AUTHOR)
+
+                # Create new branch from current head
+                new_branch = repo.create_head(pr_branch_name)
```

Comment:
I guess that this was necessary to fix the problem mentionned in https://github.com/osrf/docker_images/pull/94#issuecomment-334830221.
What I'm surprised to see is that this branch variable is not used anywhere in the code (same thing for g_branch below). Am I missing something here?

Reply:
I just wanted to capture what they returned so it would be clear that those function return specific object types, incase I have to re-read the code later. Would it be best to omit the variables?

Reply:
makes sense, in that case you can add "  # noqa" at the end of the line to make linters/test editors happy